### PR TITLE
Add UI test back for 1.0

### PR DESCRIPTION
**Pull request by Hoxbro**
Adding UI test back to branch 1.0 

Still, a work in progress and will need some fine-tuning. See `test_param.py` as an example of this.

### CODE REVIEW

File: panel/tests/ui/layout/test_card.py [hunk 1]
```diff
@@ -70,10 +70,10 @@ def test_card_not_collapsible(page, port, card_components):
     serve_panel_widget(page, port, card)
 
     # no card button to disable collapsing the card
-    card_button = page.locator('.bk.card-button')
+    card_button = page.locator('.card-button')
     expect(card_button).to_have_count(0)
     # card header and other inner objects
-    card_elements = page.locator('.bk.card > .bk')
+    card_elements = page.locator('.card > .bk')
```
Comment (mattpap) [inline panel/tests/ui/layout/test_card.py:76]: The other `.bk` needs to be dropped as well, but I'm not sure what to suggest in place of it.

Comment (Hoxbro) [reply panel/tests/ui/layout/test_card.py:76]: Yes. This PR right now is basically just a big search and replace.

**Comment by Hoxbro** [comment]
The `test_card_header_color_formatting` marked `bokeh3_failing` is because it does now work as expected. 

https://user-images.githubusercontent.com/19758978/216030779-373b348b-e11a-4634-9ef7-55a27cdca489.mp4


https://user-images.githubusercontent.com/19758978/216030804-358070df-27a5-4ce7-9ded-a9f42ad1b8dc.mp4

Another thing is that the header is collapsing when closing:

https://user-images.githubusercontent.com/19758978/216031232-1c7fd144-1963-4a15-aac0-21c1eeb86e0e.mp4

File: panel/models/data.ts [hunk 2]
```diff
@@ -6,10 +6,29 @@ export function transform_cds_to_records(cds: ColumnDataSource, addId: boolean =
...
+      for (let i = start; i < unpack_array.length; i++) {
+        if ((unpack_array[i].byteLength !== undefined) && unpack_array[i].length > 1)
+          // A typed array
+          arrays[column].push.apply(arrays[column], Array(...unpack_array[i]));
+        else
+          arrays[column].push(unpack_array[i])
+        }
+      total_length = Math.max(total_length, arrays[column].length);
+      }
+    }
```
Comment (philippjfr) [inline panel/models/data.ts:26]: Can you explain the issue? This looks like a workaround for a bokeh bug.

Comment (Hoxbro) [reply panel/models/data.ts:26]: I would classify it as a workaround, but I only see this problem on `pn.widgets.Tabulator` and not with `pn.widgets.DataFrame`. 

The problem is that when streaming data, the data sent is a typed array and not an `Array`, e.g., the following code and a `console.log(data)` at the end of the function `transform_cds_to_records`, will give the following output, when clicking on three times on the button:

image

image

This is the code I used:

``` python
import numpy as np
import pandas as pd
import panel as pn

df = pd.DataFrame(np.random.random((3, 2)), columns=["A", "B"])
widget = pn.widgets.Tabulator(df).servable()


def stream_data(*e) -> None:
    widget.stream(df)  


b = pn.widgets.Button(name="Test").servable()
b.on_click(stream_data)
```

Comment (Hoxbro) [reply panel/models/data.ts:27]: I have added a comment about it in the code. Will remove the workarounds when the upstream issue gets fixed: https://github.com/bokeh/bokeh/issues/12788

Comment (philippjfr) [reply panel/models/data.ts:27]: Would suggest removing it now and marking the streaming tests as xfail for the time being, the workaround breaks more tests than it fixes and I'd like to merge this PR asap.

Comment (Hoxbro) [reply panel/models/data.ts:27]: Will do

Comment (Hoxbro) [reply panel/models/data.ts:27]: Without removing the data workaround, no tabulator tests are failing. Would you still like for me to remove it?

The problem workaround was fixed in 1fab8bd 

```
# Unit tests
FAILED panel/tests/test_server.py::test_server_exception_handler_async_change_event[threads-synchronous_handler] - assert 0 == 1

# Examples
examples/reference/widgets/DatePicker.ipynb

# UI Test
FAILED panel/tests/ui/io/test_jupyter_server_extension.py::test_jupyter_server - AssertionError: assert '' == '0'
FAILED panel/tests/ui/io/test_jupyter_server_extension.py::test_jupyter_server_custom_resources - AssertionError: assert 'rgb(0, 0, 0)' == 'rgb(128, 0, 128)'
FAILED panel/tests/ui/io/test_jupyter_server_extension.py::test_jupyter_server_kernel_error - TimeoutError: wait_until timed out in 5000 milliseconds
FAILED panel/tests/ui/pane/test_markup.py::test_update_markdown_pane_resizes - assert 63.125 == 64
FAILED panel/tests/ui/pane/test_plotly.py::test_plotly_hover_data - AssertionError: assert None == {'points': [{'curveNumber': 0, 'pointIndex': 0, 'pointNumber': 0, 'x': 0, ...}]}
FAILED panel/tests/ui/pane/test_plotly.py::test_plotly_click_data - AssertionError: assert None == {'points': [{'curveNumber': 0, 'pointIndex': 0, 'pointNumber': 0, 'x': 0, ...}]}
FAILED panel/tests/ui/template/test_fastgridtemplate.py::test_fast_grid_template_updates - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/template/test_materialtemplate.py::test_material_template_no_console_errors - assert [<ConsoleMess... (Not Found)>] == []
FAILED panel/tests/ui/widgets/test_image.py::test_pdf_no_embed_start_page - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/widgets/test_sliders.py::test_editableslider_textinput_end[EditableFloatSlider] - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/widgets/test_sliders.py::test_editableslider_textinput_start[EditableFloatSlider] - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/widgets/test_sliders.py::test_editablerangeslider_textinput_end[EditableRangeSlider] - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
FAILED panel/tests/ui/widgets/test_sliders.py::test_editablerangeslider_textinput_start[EditableRangeSlider] - playwright._impl._api_types.TimeoutError: Timeout 20000ms exceeded.
```

Comment (philippjfr) [reply panel/models/data.ts:27]: Okay, let's leave it for now then.

**Comment by Hoxbro** [comment]
The changes made in 13d1e67 resolves https://github.com/holoviz/panel/issues/4415

**Comment by philippjfr** [comment]
Okay, we're not all the way yet but since it's passing now I'll merge.

**Pull request merged**